docs/medical-api/api-reference/document/post-upload-url.mdx

```diff
@@ -0,0 +1,126 @@
+
+Once the file is uploaded, the DocumentReference will be updated with the file attachment details on our end and uploaded it to our FHIR servers, making it available to other HIEs.
+
+## Query Params
+
+<ParamField query="patientId" type="string" required>
+  The ID of the Patient.
+</ParamField>
+
+## Body
+
+    <ParamField body="type" type="CodeableConcept" required>
+      A [CodeableConcept](/medical-api/fhir/data-types/codeableconcept) of the file type.
+      <Expandable title="Type properties">
+        <ParamField body="text" type="string" required>
+          Plain text representation of the document kind.
+        </ParamField>
+
+        <ParamField body="coding" type="Coding">
+          A [Coding](/medical-api/fhir/data-types/coding) property, which is a reference to a code defined by a terminology system.
+        </ParamField>
+      </Expandable>
+    </ParamField>
+
+    <ParamField body="description" type="string" required>
+      A brief description of the file.
+    </ParamField>
+
+    <ParamField body="context" type="DocumentReferenceContext[]" required>
+      Context of the document content.
+      <Expandable title="Context properties">
+        <ParamField body="period" type="Period" required>
+          A time [Period](/medical-api/fhir/data-types/period) property defined by a start and end date and optionally time.
+        </ParamField>
+
+        <ParamField body="facilityType" type="CodeableConcept">
+          Optional [CodeableConcept](/medical-api/fhir/data-types/codeableconcept) for the facilityType. We ask that you provide the name of the facility that generated the document under `text`.
+            <Expandable title="Facility type properties">
+          <ParamField body="type" type="CodeableConcept" required>
+            A [CodeableConcept](/medical-api/fhir/data-types/codeableconcept) for the type and name of the facility.
+              <ParamField body="text" type="string">
+                Plain text with the facility name and type.
+              </ParamField>
+
+              <ParamField body="coding" type="Coding">
+                A [Coding](/medical-api/fhir/data-types/coding) property, which is a reference to a code defined by a terminology system.
+              </ParamField>
+          </ParamField>
+            </Expandable>
+        </ParamField>
+
+      </Expandable>
+    </ParamField>
+
+## Response
+
+A presigned URL that can be used to upload a medical document to our system.
+
+```json
+{
+  "url": "https://s3.us-east-2.amazonaws.com/metriport-medical-document-uploads-staging",
+  "fields": {
+    "key": "<s3-object-key>",
+    "bucket": "<s3-bucket-name>",
+    "X-Amz-Algorithm": "AWS4-HMAC-SHA256",
+    "X-Amz-Credential": "<some-credentials-string>",
+    "X-Amz-Date": "20231020T064041Z",
+    "Policy": "<some-policy-string>",
+    "X-Amz-Signature": "<some-signature-string>"
+  }
```

Comment:
is this actually the only way to do this? we say we return a presigned URL, but that doesn't seem to be the case here

can we not just return a URL that has all of those "fields" baked-in? I think that's what we do for the Get Document URL

Reply:
I recommended that we treat `fields` as a dictionary, or `Record<string, string>`. We should not get into the implementation details of S3 on the type nor the documentation. Just need to mention that all fields present there should go on the form-data.

Reply:
> can we not just return a URL that has all of those "fields" baked-in?

No, not that I'm aware of - and we did research about this.